fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/InvertibleFunctionKeyExpression.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * InvertibleFunctionKeyExpression.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.metadata.expressions;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.RecordCoreArgumentException;
+import com.apple.foundationdb.record.logging.LogMessageKeys;
+import com.apple.foundationdb.record.metadata.Key;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * Key expression representing an invertible function. A function is invertible if it is possible to determine
+ * the pre-image of function application given the result. For example, if the function represents the squaring
+ * function, then the inverse would represent the square root. Note that this class allows the inverse to return
+ * multiple values, if there are multiple values that may map to the same output.
+ */
+@API(API.Status.EXPERIMENTAL)
+public abstract class InvertibleFunctionKeyExpression extends FunctionKeyExpression {
```

Comment:
Perhaps a couple of unit tests of this?

Reply:
Added a few to `InvertibleFunctionKeyExpressionTest`.